tests/integration/v3_lease_test.go

```diff
@@ -142,26 +143,84 @@ func TestV3LeaseGrantByID(t *testing.T) {
 	}
 }
 
-// TestRejectLeaseRequestWithInvalidID ensures leases request with a negative ID is rejected
-func TestRejectLeaseRequestWithInvalidID(t *testing.T) {
+// TestV3LeaseGrantByNegativeID ensures leases may be created by a negative given id and handled correctly
+func TestV3LeaseGrantByNegativeID(t *testing.T) {
 	integration.BeforeTest(t)
-	clus := integration.NewCluster(t, &integration.ClusterConfig{Size: 1})
+	clus := integration.NewCluster(t, &integration.ClusterConfig{Size: 3})
 	defer clus.Terminate(t)
 
-	// create lease with negative lease ID
-	gRPCAPI := integration.ToGRPC(clus.RandClient())
-	ctx := context.TODO()
-	_, err := gRPCAPI.Lease.LeaseGrant(ctx, &pb.LeaseGrantRequest{ID: -1, TTL: 1})
-	checkInvalidLeaseIDError(t, err)
-	_, err = gRPCAPI.Lease.LeaseRevoke(ctx, &pb.LeaseRevokeRequest{ID: -1})
-	checkInvalidLeaseIDError(t, err)
-	_, err = gRPCAPI.Lease.LeaseTimeToLive(ctx, &pb.LeaseTimeToLiveRequest{ID: -1})
-	checkInvalidLeaseIDError(t, err)
-}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	// grant negative leaseID
+	input := []struct {
```

Comment:
Instead of testing 3 inputs in one cluster and risk side effects (for example if first input breaks revision, error will be reported for all inputs after), could we create subtest for each input that creates it's own cluster?
Don't worry about performance, in tests correctness and clarity matters more.